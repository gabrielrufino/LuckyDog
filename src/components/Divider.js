import React from 'react';
import {StyleSheet, View} from 'react-native';

export default function Divider({direction = 'horizontal'}) {
  const styles = StyleSheet.create({
    container: {
      borderColor: '#CACAAA',
      borderBottomWidth: direction === 'horizontal' ? 1 : 0,
      borderRightWidth: direction === 'vertical' ? 1 : 0,
    },
  });

  return <View style={styles.container} />;
}
