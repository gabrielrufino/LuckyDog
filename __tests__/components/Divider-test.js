import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import Divider from '../../src/components/Divider';

describe('components/Divider.js', () => {
  it('Should renders the default divider correctly', () => {
    const tree = renderer.create(<Divider />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('Should renders the divider with vertical direction correctly', () => {
    const tree = renderer.create(<Divider direction="vertical" />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
