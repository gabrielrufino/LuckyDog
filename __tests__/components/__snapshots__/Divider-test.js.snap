// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Divider.js Should renders the default divider correctly 1`] = `
<View
  style={
    Object {
      "borderBottomWidth": 1,
      "borderColor": "#CACAAA",
      "borderRightWidth": 0,
    }
  }
/>
`;

exports[`components/Divider.js Should renders the divider with vertical direction correctly 1`] = `
<View
  style={
    Object {
      "borderBottomWidth": 0,
      "borderColor": "#CACAAA",
      "borderRightWidth": 1,
    }
  }
/>
`;
